pkg/component/component.go

```diff
@@ -360,3 +362,91 @@ func Exists(client *occlient.Client, componentName string) (bool, error) {
 	}
 	return false, nil
 }
+
+// GetComponentService returns the Service object associated with the given
+// component.
+// An error is thrown when exactly one Service is not found for the component.
+func GetComponentService(client *occlient.Client, componentName string, applicationName string) (*corev1.Service, error) {
```

Comment:
I'm not sure that this function belongs here.  If you really want to keep it here than it definitely shouldn't be public. Otherwise, it is a potential risk for creating cyclic imports like in the `storage` :-(

It might be better to do it similarly like here: https://github.com/redhat-developer/odo/pull/352/commits/f2d5b10ca01fec612f7349e606ba3966ea107574#diff-6025e9b620e9321c03c11bf29e5c4dbdR1190

Reply:
moved this similar to the current deployment config logic